### PR TITLE
Refactoring leftovers of unit tests for Jobs/Datasets.

### PR DESCRIPTION
**Pull request by stepanzharychevbroadcom**
## Proposed changes

Refactoring leftovers of unit tests for Jobs/Datasets.

Milestone: 1.7

Changelog:
Moved leftovers of tests from extension.ts to dataset/actions.ts
Moved leftovers of tests from extension.ts to job/actions.ts

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] `npm run vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: __mocks__/mockCreators/jobs.ts [hunk 1]
```diff
@@ -68,9 +87,11 @@ export function createJobsTree(session: imperative.Session, iJob: zowe.IJob, pro
     };
     testJobsTree.mSessionNodes = [];
     testJobsTree.mSessionNodes.push(jobNode);
-    testJobsTree.addFavorite.mockImplementation((newFavorite) => { testJobsTree.mFavorites.push(newFavorite); });
+    testJobsTree.addFavorite.mockImplementation((newFavorite) => {
+        testJobsTree.mFavorites.push(newFavorite);
```
Comment (katelynienaber) [inline __mocks__/mockCreators/jobs.ts:91]: Why did you move this to 3 lines? Seems cleaner to have it on 1

Comment (katelynienaber) [inline __mocks__/mockCreators/jobs.ts:91]: Why did you move this to 3 lines? Seems cleaner to have it on 1

File: __tests__/__unit__/dataset/actions.unit.test.ts [hunk 2]
```diff
@@ -1557,3 +1561,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
...
+        blockMocks.testDatasetTree.getHistory.mockReturnValue([]);
+        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profileInstance);
+        mocked(vscode.window.showInputBox).mockResolvedValue("test");
+        const createDataSetSpy = jest.spyOn(blockMocks.mvsApi, "createDataSet");
+        createDataSetSpy.mockReset();
+        const getChildrenSpy = jest.spyOn(blockMocks.datasetSessionNode, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
```
Comment (katelynienaber) [inline __tests__/__unit__/dataset/actions.unit.test.ts:1605]: Only 1 call to the tested function per it() block?

File: __tests__/__unit__/dataset/actions.unit.test.ts [hunk 3]
```diff
@@ -1557,3 +1561,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
...
+        mocked(vscode.window.showInputBox).mockResolvedValue("test");
+        const createDataSetSpy = jest.spyOn(blockMocks.mvsApi, "createDataSet");
+        createDataSetSpy.mockReset();
+        const node = new ZoweDatasetNode("HLQ.TEST.TO.NODE", vscode.TreeItemCollapsibleState.None, null, blockMocks.sessionWithoutCredentials);
+        node.contextValue = globals.DS_SESSION_CONTEXT;
+        const getChildrenSpy = jest.spyOn(node, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
```
Comment (katelynienaber) [inline __tests__/__unit__/dataset/actions.unit.test.ts:1644]: 1 call to tested function per it() block.

File: __tests__/__unit__/dataset/actions.unit.test.ts [hunk 4]
```diff
@@ -1557,3 +1561,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
...
+        mocked(vscode.window.showInputBox).mockResolvedValue("test");
+        const createDataSetSpy = jest.spyOn(blockMocks.mvsApi, "createDataSet");
+        createDataSetSpy.mockReset();
+        const node = new ZoweDatasetNode("HLQ.TEST.TO.NODE", vscode.TreeItemCollapsibleState.None, null, blockMocks.sessionWithoutCredentials);
+        node.contextValue = globals.DS_SESSION_CONTEXT + globals.FAV_SUFFIX;
+        const getChildrenSpy = jest.spyOn(node, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
```
Comment (katelynienaber) [inline __tests__/__unit__/dataset/actions.unit.test.ts:1683]: 1 call per it() block

Comment (katelynienaber) [inline __tests__/__unit__/dataset/actions.unit.test.ts:1683]: Only 1 call to tested function per it() block?? Also occurs in next 2 describe() blocks.

File: __mocks__/mockCreators/jobs.ts (hunk 1 above)
Comment (katelynienaber) [inline __mocks__/mockCreators/jobs.ts:91]: Why did you move this to 3 lines? Seems neater to have it on 1.

File: __mocks__/mockCreators/jobs.ts [hunk 5]
```diff
@@ -44,6 +44,25 @@ export function createIJobObject(): zowe.IJob {
     };
 }
 
+export function createIJobFileObject(): zowe.IJobFile {
```
Comment (katelynienaber) [inline __mocks__/mockCreators/jobs.ts:47]: This is going to conflict with my ZosJobsProvider.unit.test.ts refactoring :(

File: __tests__/__unit__/dataset/actions.unit.test.ts (hunk 4 above)
Comment (stepanzharychevbroadcom) [reply __tests__/__unit__/dataset/actions.unit.test.ts:1683]: Answering on all at once, specific of this test is verification of proper run with different types without details, splitting it will result in hundreds of lines of the duplicates. I think that's an exception which can be made, if no - fine, I will spend time splitting it to 5 test cases with duplicates.

Comment (katelynienaber) [reply __tests__/__unit__/dataset/actions.unit.test.ts:1683]: Yeah, I guess that's fine. I'm just being OCD about it...

File: __mocks__/mockCreators/jobs.ts (hunk 1 above)
Comment (VitGottwald) [reply __mocks__/mockCreators/jobs.ts:91]: That is likely webstorm code formatting - similar to other changes in this PR. Once we get prettier we will not have this kind of changes in PRs.

File: __tests__/__unit__/dataset/actions.unit.test.ts [hunk 6]
```diff
@@ -1584,3 +1588,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
...
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-Classic");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Partitioned" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PDS");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Sequential" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PS");
```
Comment (VitGottwald) [inline __tests__/__unit__/dataset/actions.unit.test.ts:1649]: These area really 5 different tests. It would make sense to separate them. But since @stepanzharychevbroadcom has moved to another group and this is an improvement over what we had, lets leave it like this and restructure it in the future.

Comment (katelynienaber) [reply __tests__/__unit__/dataset/actions.unit.test.ts:1649]: Stepan's response to the issue was as follows:

> Answering on all at once, specific of this test is verification of proper run with different types without details, splitting it will result in hundreds of lines of the duplicates. I think that's an exception which can be made, if no - fine, I will spend time splitting it to 5 test cases with duplicates.

Comment (VitGottwald) [reply __tests__/__unit__/dataset/actions.unit.test.ts:1649]: I think that is a bit of exaggerated. The setup logic can be share in a describe block or extracted to function and reused.
